src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs

```diff
@@ -113,5 +116,8 @@ private static IViewEngine ResolveViewEngine(ViewComponentContext context)
         {
             return context.ViewContext.HttpContext.RequestServices.GetRequiredService<ICompositeViewEngine>();
         }
+
+        private static bool IsApplicationRelativePath(string path) =>
+            path[0] == '~' || path[0] == '/';
```

Comment:
Suggest inlining into the condition above. As-is looks like a source of `IndexOutOfRangeException`s.